app/src/main/java/com/codesoom/assignment/server/TodoServerHandler.java

```diff
@@ -0,0 +1,80 @@
+import com.codesoom.assignment.infrastructure.mapping.PutHttpMapping;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class TodoServerHandler implements HttpHandler {
+    public final AppConfig appConfig;
+
+    public TodoServerHandler(AppConfig appConfig) {
+        this.appConfig = appConfig;
+    }
+
+    /*
+        ToDo request 로그 찍기
+     */
+    @Override
+    public void handle(HttpExchange exchange)  {
+        try {
+            String path = exchange.getRequestURI().getPath();
+            String method = exchange.getRequestMethod().toLowerCase(Locale.ROOT);
+            String requestBody = new BufferedReader(
+                    new InputStreamReader(exchange.getRequestBody())
+            )
+                    .lines().collect(Collectors.joining("\n"));
+
+            Map<String, Function<TodoService, HttpMapping>> httpMethodToConstructorMap = getHttpMethodToConstructorMap();
+            HttpRequest httpRequest = toHttpRequest(path, method, requestBody);
+            HttpMapping httpMapping = httpMethodToConstructorMap.get(method).apply(appConfig.todoService());
+            HttpResponse httpResponse = httpMapping.process(httpRequest);
+            writeHttpResponse(exchange, httpResponse);
+        }catch (IOException e){
+            System.out.printf("in handle -> {}",e.getMessage());
+        }
+
+    }
+
+    private HttpRequest toHttpRequest(String path, String method, String requestBody) {
+        return new HttpRequest.HttpRequestBuilder()
+                .path(path)
+                .method(method).requestBody(requestBody).builder();
+    }
+
+    private Map<String, Function<TodoService, HttpMapping>> getHttpMethodToConstructorMap() {
+        Map<String, Function<TodoService, HttpMapping>> httpMethodToConstructorMap = new HashMap<>();
+        httpMethodToConstructorMap.put("get", GetHttpMapping::new);
+        httpMethodToConstructorMap.put("post", PostHttpMapping::new);
+        httpMethodToConstructorMap.put("put", PutHttpMapping::new);
+        httpMethodToConstructorMap.put("delete", DeleteHttpMapping::new);
+        return httpMethodToConstructorMap;
+    }
+
+    private void writeHttpResponse(HttpExchange httpExchange, HttpResponse httpResponse) throws IOException {
+        httpExchange.sendResponseHeaders(httpResponse.getStatusCode(), httpResponse.getBody().getBytes().length);
+        OutputStream outputStream = httpExchange.getResponseBody();
+        outputStream.write(httpResponse.getBody().getBytes());
+        outputStream.flush();
+        outputStream.close();
+    }
```

Comment:
```suggestion
    private void writeHttpResponse(HttpExchange httpExchange, HttpResponse httpResponse) {
        try (OutputStream outputStream = httpExchange.getResponseBody()) {
            var body = httpResponse.getBody().getBytes();
            httpExchange.sendResponseHeaders(httpResponse.getStatusCode(), body.length);
            outputStream.write(body);
            outputStream.flush();
        } catch (IOException e){
            System.out.printf("in handle -> {}",e.getMessage());
        }
    }
```

한 번 다음처럼 바꿔보았어요!
try-catch 문이 너무 크게 잡혀있으면 어느 부분 때문에 `IOException` 을 잡아야 하는지 알기 힘들 수가 있어요!
또한 `OutputStream` 은 `Closeable` 을 구현하고 있어서 `try-with-resources statement` 를 이용하면 예외가 나건 안나건 자동으로 닫아져요! 

기존 코드는 중간에 예외가 나면 잘 닫히지 않을 수 있어요!

Reply:
감사합니다!!
IO  스트림 관련 클래스의 예외처리할때는 try-with-resources statement를 사용해서 자원 반납을 자동으로 하고 

예외를 잡을때는 큰 범위가 아닌 꼭 필요한 범위를 잡도록 하겠습니다!